lib/magma/censor.rb

```diff
@@ -0,0 +1,41 @@
+class Magma
+  class Censor
+    attr_reader :model, :record_name
+    def initialize(user, project_name)
+      @user = user
+      @project_name = project_name
+    end
+
+    def censored?(model, revisions)
```

Comment:
This is a super awkward interface.  It uses `?` notation to suggest it is a predicate, but it returns nil in all cases, and instead yields strings indicated reasons for censorship.

Would it be possible to change this to two functions:

```
def censored?
  censored_reasons.length > 0
end

def censored_reasons
 result = []
 return result unless restrict?
 result << *restricted_identifiers
 ...
 result
end
```

I feel this would still be easy to test using the `includes?` matchers and it would be a bit more obvious an interface.  You can even pump arrays into each other pretty simple ala `@errors << *censored_reasons`

Reply:
It would also make it possible to cache the result via an attribute `||=` operation, where as that is not possible with a block yield.